keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -26,14 +29,15 @@ import org.springframework.stereotype.Component
  */
 @Component
 class ApplicationService(
-  private val repository: KeelRepository,
-  private val artifactRepository: ArtifactRepository
+  private val resourceRepository: ResourceRepository,
+  private val artifactRepository: ArtifactRepository,
+  private val deliveryConfigRepository: DeliveryConfigRepository
```

Comment:
Little side-note, but wanted to call your attention to this class (and potentially a pattern) here. Personally, I really feel like lumping all the various repository methods into a giant interface like `KeelRepository` is not the best approach. Not only does it make that interface and the implementation extremely hard to read and understand, but it violates the single responsibility principle by adding business logic into what, by definition, should be a class concerned exclusively with persistence. 

The approach taken here, instead, is to have an object whose specific and declared purpose is to be a service object responsible for that kind of business logic. Upstream callers, like the `ApplicationController` in this case, call into the high-level methods presented by this class to abstract details related to calling multiple persistence repositories and adding additional processing of the data.

I would propose that we reconsider the combined repository approach in favor of purpose-built service objects for specific sub-areas of the domain model.

Reply:
But, the `KeelRepository` class is meant to deal with business logic - i.e. how do the objects relate, what order must things be done in, what happens when we delete one thing. Without a way to deal with all of the objects at once (delivery config, artifacts, resources) we will have consistency and concurrency bugs. We should talk about this offline.

For now, please use the `KeelRepository` here.

Reply:
You could in fact think of `KeelRepository` as a purpose built service object that itself is concerned with persistence so that things like the `ApplicationController` don't have to be.

Reply:
Let's take it offline. I'll open an issue and transfer these comments there. I didn't want to loose the thought but should have done that in the first place. My bad.

Reply:
Opened #927